From df594b97149efe1522a40d909023136d4940d717 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:27 -0400
Subject: [tools] lib/traceevent: Get rid of malloc_or_die() in trace_seq_init()

Message-id: <1408953210-26343-260-git-send-email-jolsa@redhat.com>
Patchwork-id: 88899
O-Subject: [PATCH RHEL7.1 BZ1133083 259/322] tools lib traceevent: Get rid of malloc_or_die() in trace_seq_init()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 504586e0954bcf9550dfdea37d3234174ed1d68f
Author: Namhyung Kim <namhyung@kernel.org>
Date: Wed Jan 15 10:45:26 2014 +0900

description
===========
Use plain malloc() and check its return value.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/trace-seq.c b/tools/lib/traceevent/trace-seq.c
index e454a2c..ec3bd16 100644
--- a/tools/lib/traceevent/trace-seq.c
+++ b/tools/lib/traceevent/trace-seq.c
@@ -57,8 +57,11 @@ void trace_seq_init(struct trace_seq *s)
  s->len = 0;
  s->readpos = 0;
  s->buffer_size = TRACE_SEQ_BUF_SIZE;
- s->buffer = malloc_or_die(s->buffer_size);
- s->state = TRACE_SEQ__GOOD;
+ s->buffer = malloc(s->buffer_size);
+ if (s->buffer != NULL)
+  s->state = TRACE_SEQ__GOOD;
+ else
+  s->state = TRACE_SEQ__MEM_ALLOC_FAILED;
 }
 
 /**
-- 
1.7.1